Governance: Add TOC description (#510)
pingcap/community: TOC/README.md

# TiDB Community Technical Oversight Committee (TOC)

The TiDB Community TOC serves as the main bridge and channel for coordinating and information sharing across companies and organizations. It is the coordination center for solving problems in terms of resource mobilization, technical research and development direction in the current community and cooperative projects.

## Responsibilities:

- Partner cooperation project progress monitoring and guidance
- Supervision and guidance on the progress of the TiDB ecological project

> - Participate in technical route discussion and decision-making, such as version release and roadmap formulation, etc.

Reviewer: > version release

@winkyao @sunxiaoguang please make sense whether or not it is the case. I don't think TOC will currently or later take responsibility of releases.

Reviewer 2: You are right, I overlooked this rule yesterday. I personally think it is not necessary to plan release in TOC.

Reviewer 3: You are right @tisonkun

Author: OK, I will delete the version release.

- Open source collaboration process discussion and decision
- Discussion and decision-making of community-related activities

## Membership

TOC members include official members and non-voting members:

- Official members have the right to participate in TOC meetings, propose proposals, and vote.
- Non-voting members have the right to participate in TOC meetings, propose proposals, but do not have the right to vote.

> TOC members come from community partners who have signed a cooperation agreement with PingCAP and representatives of important community projects. Each community partner has 1 official member seat. TiDB, TiKV and Chaos Mesh®  each have 1 official member seat. The seats of non-voting members are not limited.

Reviewer: > who ... with PingCAP _**and**_ representatives....

"and" or "or"?

Reviewer: How many seats does PingCAP have?

Author: 1. The subject TOC members is plural, I think we should use "and" ?Due to Xiaoguang’s job change, PingCAP currently has 2 seats,

Reviewer 3: I think @sunxiaoguang is the chair of the TOC, not represent the PingCAP, he just plays the role in the community.

Author: I discussed with wink, we decide to temporarily remove the limit on the number of seats and modify the wording of the TOC membership in the table.

## Nomination

> All official members of TOC have the right to nominate. Nominees are required to meet the following standards:

Reviewer: > All official members of TOC have the right to nominate

what to nominate?

Reviewer 2: Hm, it's better to make it clear this is for new member nomination.

Author: nominata representative as new TOC member.

> - He/she must have enough energy to invest in TOC affairs, such as participating in TOC discussions and [decision making](./Proposal), and participating in TOC monthly meetings.

Reviewer: You can use "The nominee" rather than "He/she"

Author: OK, thanks.

> - He/she must have a deep understanding of TiDB or related ecological technologies. He/she is at least the Committer of TiDB/TiKV/Chaos Mesh®  or TiDB ecological project.

Reviewer: ```suggestion
- The nominee must have a deep understanding of TiDB or other projects in the ecosystem.
- The nominee must be at least a committer of TiDB, TiKV, Chaos Mesh®, or other projects in the ecosystem.
```

We'd better have a table on "projects in ecosystem".

Reviewer: And I'm afraid in this rules, "Yuqing Bai" will be unsatisfied.

Reviewer 2: I had similar question and specially discussed with Xiaolu about this. The other projects in the ecosystem includes projects like TiBigData or Weir. But I agree that even with these projects included. It's still not clear how to define 'deep understanding'.

> - The nominee must have a deep understanding of TiDB or related ecological technologies. The nominee is at least the Committer of TiDB/TiKV/Chaos Mesh®  or TiDB ecological project.

Reviewer 2: After having a second thought about this particular standard, I suggest we eliminate the prerequsite that the nominee must be a committer. The rationale behind this suggestion are: 

1. Members of TOC represents the organization that have key impact to the community, but not for themselves.
2. Their responsibilities are mainly to coordinate resource investment and make decisions for long-term goals. These responsibilities do not require writing lots of code but having greater vision about the TiDB community.

Author: Yes, you convinced me. If no one objects, I will eliminate the committer prerequisite.

> - He/she must belong to a certain organization of TOC membership and be able to participate in TOC on behalf of his/her organization.

Reviewer: It makes no sense for "representatives of important community projects". For example, "Jinpeng Zhang" for TiKV is on behalf of "PingCAP"?

Author: I think Jinpeng is the representative of TiKV, and he joined TOC as TiKV Maintainer. cc @winkyao

Reviewer 3: Yes. @luzizhuo I think Jinpeng is the TiKV Maintainer representative.

| Jinpeng Zhang zhangjinpeng@pingcap.com | [zhangjinpeng1987](https://github.com/zhangjinpeng1987) | TiKV | TOC Official Member |
| Chengwen Yin cwen@pingcap.com | [cwen0](https://github.com/cwen0) | Chaos Mesh®  | TOC Official Member |
| Dian Luo luodian@pingcap.com | [andylokandy](https://github.com/andylokandy) | TiKV | TOC Non-voting Member |
| Jay Li jay@pingcap.com | [BusyJay](https://github.com/BusyJay) | TiKV | TOC Non-voting Member |
| Cong Wang wangcong@pingcap.com | [bb7133](https://github.com/bb7133) | TiDB | TOC Non-voting Member |

### Emeritus Members

TOC members has no concept of tenure, but will retire under the following circumstances:

- Actively choose to retire due to personal reasons.
- When he/she can no longer participate in TOC affairs on behalf of this organization due to leaving or withdrawing from the organization they represent.
- The organization he/she represents no longer exists.

Retirees will become emeritus members of TOC. The list of emeritus members is as below.

- Xianlin Chen，Head of Technology Hub at PalFish
- Qiang Zhou，Chaos Mesh® Maintainer

## TOC Meeting

> The TOC monthly online meeting is held at 8 pm on the last Friday of each month. Meeting invitations will be sent through the TOC mail group: https://lists.tidb.io/g/toc. 

Reviewer: 8pm should be with timezone

Author: OK, thanks.

If official members cannot attend the monthly meeting for special reasons, they must ask for leave in advance.

### Meeting Minutes

[English version](https://docs.google.com/document/d/1m9Uk-JkqO_KaSZaLjL47hW1L_2Bp-yW1StCuGQzOFTk/edit)

[Chinese version](https://docs.google.com/document/d/1KvavvwWpABbTEJO3IMLBykGMOUMq4soL2ityDryDqgw/edit)

## Decision Making
Matters that require TOC members to make decisions can be proposed at the TOC meeting or indicated by a pull request adding an entry under the [Proposal folder](./Proposal).

All TOC members have the right to make proposals. Only official members have the right to vote, and each has 1 vote. Decisions are made by lazy consensus. If anyone votes against it, discussion is needed. If no agreement can be reached, Majority Vote will be used.

| Actions | Description | Approval | Binding Voters |  Minimum Length (days) |
|:----|:----|:----|:----|:----|

> | New TOC Member |  When a new TOC member is nominated.  |  [Lazy Consensus](./Proposal/voting-rules.md#lazy-consensus)  |  TOC Official Members  |  6  |

Reviewer: You said 2 days in `voting-rules.md`

> For all votes, voting must be open for at least 2 days.

Author: at least 2 days.

Reviewer: here you wrote Minimum Length 6 days. These two information confict.

Author: The rule "at least 2 days" is for all votes. As long as the voting time is more than 2 days, it is in line with the rules. 
Here the vote "New TOC Member" must last for at least 6 days. They don't conflict.

Reviewer: Make sense. Thanks for your clarification.
